pkg/cli/images.go

```diff
@@ -76,16 +77,55 @@ func (a *Image) Run(cmd *cobra.Command, args []string) error {
 	})
 
 	for _, image := range images {
-		if image.Tag == "" && image.Repository == "" && !a.All {
-			continue
+		imagePrint := imagePrint{
+			TypeMeta:   image.TypeMeta,
+			ObjectMeta: image.ObjectMeta,
+			Digest:     image.Digest,
+			Tag:        "",
+			Reference:  "",
+		}
+		for _, tag := range image.Tags {
+			if tag == "" && !a.All {
+				continue
+			}
+			if includesTag(tag) {
+				imagePrint.Tag = strings.Split(tag, ":")[1]
+			} else if includesRepoAndTag(tag) {
+				imagePrint.Tag = strings.Split(strings.Split(tag, "/")[len(strings.Split(tag, "/"))-1], ":")[1]
+			} else {
+				imagePrint.Tag = "latest"
+			}
+
+			imagePrint.Reference = tag
+			imagePrint.Repository = parseRepo(tag)
+			out.Write(imagePrint)
+		}
+		if len(image.Tags) == 0 && a.All {
+			out.Write(imagePrint)
 		}
-		out.Write(image)
 	}
 
 	return out.Err()
 }
 
+type imagePrint struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	Digest            string `json:"digest,omitempty"`
+	Repository        string `json:"repository,omitempty"`
```

Comment:
dont need repository, reference, clusterName here.

I am fairly certain you can get rid of typeMeta as well, since you aren't using it for printing. ObjectMeta contains the Name field, so you kind of need that for `{"Image-ID", "{{trunc .Name}}"},`, but you could drop ObjectMeta and just replace with just a `Name string` field

Reply:
removed